packages/core/src/pages/[...slug].tsx

```diff
@@ -180,7 +79,18 @@ export const getStaticProps: GetStaticProps<
   { slug: string[] },
   Locator
 > = async ({ params, previewData }) => {
-  const [{ data, errors = [] }, page, globalSections] = await Promise.all([
+  const [landingPage, globalSections] = await Promise.all([
+    getLandingPageBySlug(params?.slug[0], previewData),
+    getGlobalSectionsData(previewData),
+  ])
+
+  if (landingPage) {
```

Comment:
I think we can improve this bit of code by only awaiting when needed, as the getGlobalSectionsData may endup blocking the collection page load.

Consider the scenario where `getLandingPageBySlug` returns way before `getGlobalSectionsData`. It could be that there's no landing page, so there's no need to wait for `getGlobalSectionsData` to finish to then start fetching the collection page.

Awaiting them separately only when they're used may yield better results in this case.

Reply:
I improved this behavior, thank you for the tips and the explanation!